apps/web/src/app/pages/trade/bridge/types.ts

```diff
@@ -178,3 +183,30 @@ export class XToken extends Token {
 }
 
 export type XTokenMap = { [key: string | number]: XToken };
+
+export type XWalletAssetRecord = {
+  baseToken: Token;
+  xTokenAmounts: { [key in XChainId]: CurrencyAmount<Currency> | undefined };
+  isBalanceSingleChain: boolean;
+  total: BigNumber;
+  value: BigNumber | undefined;
+};
+
+export type Position = {
+  collateral: CurrencyAmount<Currency> | undefined;
+  loan: BigNumber;
+  isPotential?: boolean;
+};
+
+export type XPositionsRecord = {
```

Comment:
I think `XPositionRecord` and `XCollateral` are logically same types. Unify the two types into one type. and refactor the code using the type.

Reply:
isPositionSingleChain == isCollateralSingleChain

Reply:
positions == chains & total. 
you can rename `chains` in `XCollateral` to `positions` and add `total` in the `positions` type. in that case, key is icon xChainId.

Reply:
moved to https://github.com/balancednetwork/balanced-network-interface/issues/1388